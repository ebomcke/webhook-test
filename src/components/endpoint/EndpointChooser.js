import React from 'react';
import { compose } from 'recompose';
import { withAccount } from '../../contexts/Account';
import { Dropdown } from 'semantic-ui-react';
import { withEndpoints } from '../../contexts/Endpoints';

const EndpointChooser = ({ account, endpoints, onChange }) => (
  <Dropdown
    selection
    search
    options={endpoints.map(endpoint => ({
      text: `/${account.organisationName}/${endpoint.path}`,
      value: endpoint.id,
    }))}
    onChange={(event, data) =>
      onChange(endpoints.find(e => e.id === data.value))
    }
  />
);
export default compose(
  withAccount,
  withEndpoints,
)(EndpointChooser);
