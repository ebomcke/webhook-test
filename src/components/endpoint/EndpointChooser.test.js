import React from 'react';
import { create } from 'react-test-renderer';
import EndpointChooser from './EndpointChooser';
import endpoints from '../../contexts/Endpoints';
import { Dropdown } from 'semantic-ui-react';

const onChange = jest.fn();

describe('Rendering tests', () => {
  test('Basic', () => {
    const endpointChooser = create(<EndpointChooser />);
    expect(endpointChooser.toJSON()).toMatchSnapshot();
  });
});

describe('Interactions tests', () => {
  afterEach(() => {
    onChange.mockClear();
  });

  test('Selecting an endpoint calls onchange with the right endpoint object', () => {
    const endpointChooser = create(<EndpointChooser onChange={onChange}/>);
    const dropdown = endpointChooser.root.findByType(Dropdown);
    dropdown.props.onChange(null, {
      value: endpoints[0].id,
    });
    expect(onChange.mock.calls.length).toBe(1);
    expect(onChange.mock.calls[0][0]).toEqual(endpoints[0]);
  });
});
