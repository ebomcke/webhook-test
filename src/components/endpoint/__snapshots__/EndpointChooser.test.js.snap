// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering tests Basic 1`] = `
<div
  aria-expanded={false}
  className="ui search selection dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="combobox"
>
  <input
    aria-autocomplete="list"
    autoComplete="off"
    className="search"
    onChange={[Function]}
    style={
      Object {
        "width": undefined,
      }
    }
    tabIndex={0}
    type="text"
    value=""
  />
  <div
    aria-live="polite"
    className="text"
    role="alert"
  />
  <i
    aria-hidden="true"
    className="dropdown icon"
    onClick={[Function]}
  />
  <div
    className="menu transition"
    role="listbox"
  >
    <div
      aria-checked={false}
      aria-selected={true}
      className="selected item"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        /unit-test-org/unit-test-1
      </span>
    </div>
    <div
      aria-checked={false}
      aria-selected={false}
      className="item"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        /unit-test-org/unit-test-2
      </span>
    </div>
  </div>
</div>
`;
